### PR TITLE
feature(auth): add simple login with chrome identity

### PR DESCRIPTION
**Pull request by Wuesteon**

### CODE REVIEW

File: src/components/Login.vue [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+  })
+})
+
+defineExpose({
+  isAuthenticated,
+  userProfile
+})
+</script>
+
+<style scoped>
```
Comment (robert197) [inline src/components/Login.vue]: We have tailwind. We don't need the css

File: src/background/index.ts [hunk 2]
```diff
@@ -5,6 +5,55 @@ chrome.runtime.onInstalled.addListener((): void => {
   console.log('Extension installed')
 })
 
+chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
+  if (request.action === 'login') {
+    const clientId = '34932676923-luth1cmkkhoqhq3kaoealqj2poosaln4.apps.googleusercontent.com';
```
Comment (robert197) [inline src/background/index.ts]: This has to be an variable which we can change by environment. 

When we release the extension, we will probably have differernt client ID and we might will have a staging environment. How do we want to deal with it?

File: src/components/Login.vue [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+    userProfile.value = {
+      name: data.name,
+      email: data.email,
+      picture: data.picture
+    }
+
+    // Store user profile
+    chrome.storage.local.set({ userProfile: userProfile.value })
+  } catch (err) {
+    console.error('Error fetching user profile:', err)
```
Comment (robert197) [inline src/components/Login.vue:97]: We should also show some error message on the frontend here.

File: src/components/Login.vue [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+onMounted(async () => {
+  chrome.storage.local.get(['accessToken', 'userProfile'], (result) => {
+    if (result.accessToken) {
+      isAuthenticated.value = true
+      userProfile.value = result.userProfile || null
+    }
+  })
+})
+
+defineExpose({
```
Comment (robert197) [inline src/components/Login.vue]: I think we do not need `defineExpose` here?

File: src/manifest.ts [hunk 5]
```diff
@@ -34,13 +41,31 @@ export async function getManifest(): Promise<Manifest.WebExtensionManifest> {
...
       'storage',
       'activeTab'
     ],
+    host_permissions: [
+      'http://*/*',
+      'https://*/*',
+      'http://localhost:1991/*'
+    ],
+    oauth2: {
+      client_id: '34932676923-luth1cmkkhoqhq3kaoealqj2poosaln4.apps.googleusercontent.com',
```
Comment (robert197) [inline src/manifest.ts:59]: We have the client_id here as well. can we centralise it and make sure that we can support multiple environments?

File: src/components/Login.vue (hunk 4 above)
Comment (robert197) [reply src/components/Login.vue]: Ok i just swa in Popup.vue tthe `isAuthenticated` is used. But we won't be able to use this on the settings page.

in order to share the variable, between the popup and the settings page, you need to use the stuff in `~/logic/storage`

File: src/manifest.ts (hunk 5 above)
Comment (robert197) [reply src/manifest.ts:59]: Has been discussed, we want to tackle this later. 

Ticket has been crated.

File: src/assets/google.svg
Comment (robert197) [inline src/assets/google.svg:1]: We have now the svg in both places.

public/google.svg and src/assets/google.svg

I guess this should be only in one place.

Comment (Wuesteon) [reply src/assets/google.svg:1]: done

File: src/components/Login.vue (hunk 3 above)
Comment (Wuesteon) [reply src/components/Login.vue:97]: displayed in ui

**Pull request merged**